File: formal/coqimp/task05/extraction_hs/RecordSet.hs
module RecordSet where

import qualified Prelude

type Reader e t = e -> t

constructor :: a2 -> Reader a1 a2
constructor x _ =
  x

type Setter r t = (t -> t) -> r -> r
  -- singleton inductive, whose constructor was Build_Setter
  
set :: (a1 -> a2) -> (Setter a1 a2) -> (a2 -> a2) -> a1 -> a1
set _ setter =
  setter

File: formal/coqimp/task01/extraction_ml/RecordSet.ml

type ('e, 't) coq_Reader = 'e -> 't

(** val constructor : 'a2 -> ('a1, 'a2) coq_Reader **)

let constructor x _ =
  x

type ('r, 't) coq_Setter = ('t -> 't) -> 'r -> 'r
  (* singleton inductive, whose constructor was Build_Setter *)

(** val set : ('a1 -> 'a2) -> ('a1, 'a2) coq_Setter -> ('a2 -> 'a2) -> 'a1 -> 'a1 **)

let set _ setter =
  setter
